pandas/core/reshape/pivot.py

```diff
@@ -683,3 +697,35 @@ def _get_names(arrs, names, prefix="row"):
             names = list(names)
 
     return names
+
+
+def _build_names_mapper(names, shared_col_row_names, suffix):
+    keys, counts = _value_counts_arraylike(names, dropna=False)
+    names_count = dict(zip(keys, counts))
+
+    names_mapper = {}
+    unique_names = []
+    # We reverse the names so the numbers are in the order given by the user
+    for name in reversed(names):
+        mapped_name = name
+        name_count = names_count[name]
+
+        # Adds a number if name is duplicated within columns/rows
+        if name_count > 1:
+            mapped_name = "{mapped_name}_{name_count}".format(
+                mapped_name=mapped_name, name_count=name_count
+            )
```

Comment:
i think we do something like this in some of the groupby code.  possibly "uniqueify"?

Reply:
Are you referring to _make_unique(seq)? That one deduplicates lambda functions, in this case we care more about the naming